Assignment7/script.js

```diff
@@ -0,0 +1,315 @@
+        validateZipCode()
+    }
+
+    if (!(errorMessage == "")) {
+        alert(errorMessage);
+        document.myform.focus();
+        return false;
+    }
+    return true;
+}
+
+function validateFirstName() {
+
+    var re = /^[A-Za-z]+$/;
+    var firstName = document.getElementById("first-name").value;
+
+    if ((firstName.length > 2 && re.test(firstName))) {
+        return true;
+    }
+    else {
+        errorMessage = errorMessage + "First Name : Please enter valid first name.\n";
+        return false;
+    }
+
+}
+
+function validateLastName() {
+
+    var re = /^[A-Za-z]+$/;
+    var lastName = document.getElementById("last-name").value;
+
+    if ((lastName.length > 2 && re.test(lastName))) {
+        return true;
+    }
+    else {
+        errorMessage = errorMessage + "Last Name : Please enter valid last name.\n";
+        return false;
+    }
+
+}
+
+function validateEmail() {
+
+    var re = /^([A-Za-z0-9_\-\.])+\@([A-Za-z0-9_\-\.])+\.([A-Za-z]{2,4})$/;
+    var email = document.getElementById("email").value;
+
+    if (re.test(email)) {
+        return true;
+    }
+    else {
+        errorMessage = errorMessage + "Email : Please enter valid email address.\n";
+        return false;
+    }
+
+}
+
+function validatePhone() {
+
+    var contact = document.getElementById("phone").value;
+
+    if (contact.length > 8 && (!isNaN(contact))) {
+        return true;
+    }
+    else {
+        errorMessage = errorMessage + "Contact : Please enter valid contact number.\n";
+        return false;
+    }
+
+}
+
+function validateZipCode() {
+
+    var zipcode = document.getElementById("zipcode").value;
+
+    if (zipcode.length == 6 && (!isNaN(zipcode))) {
+        return true;
+    }
+    else {
+        errorMessage = errorMessage + "ZipCode : Please enter valid ZipCode.\n";
+        return false;
+    }
+
+}
+
+function getProfileData() {
+    var urlString = window.location;
+    var url = new URL(urlString);
+    var state = url.searchParams.get("state");
+
+    document.getElementById("name").innerHTML = url.searchParams.get("FirstName") + " " + url.searchParams.get("LastName");
+    document.getElementById("email").innerHTML = url.searchParams.get("Email");
+    document.getElementById("city").innerHTML = url.searchParams.get("city");
+    document.getElementById("phone").innerHTML = url.searchParams.get("phone");
+    document.getElementById("address").innerHTML = url.searchParams.get("address");
+    document.getElementById("state").innerHTML = state;
+
+    if (state == "Haryana") {
+        document.getElementById("label-1").innerHTML = "Zipcode:";
+        document.getElementById("label-1-content").innerHTML = url.searchParams.get("zipcode");
+        document.getElementById("label-2").innerHTML = "Have Hosting:";
+        document.getElementById("label-2-content").innerHTML = url.searchParams.get("hosting");
+    }
+    if (state == "Rajasthan") {

```

Comment:
Use else if instead of if when it is not necessary to run all the conditions. It may cause performance issues.